MacFSEvents

pbronez opened the pull request:
Conda recipe for MacFSEvents ([PyPi](https://pypi.python.org/pypi/MacFSEvents), [Github](https://github.com/malthe/macfsevents)), a "Thread-based interface to file system observation primitives" for OS X.

Builds successfully on my Macbook with:

```bash
conda build \
--channel https://conda.anaconda.org/conda-forge/noarch \
--channel https://conda.anaconda.org/conda-forge/osx-64 \
--override-channels \
recipes/macfsevents/meta.yaml
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/macfsevents```) and found some lint.

Here's what I've got...


For **recipes/macfsevents**:

 * The recipe must have some tests.
 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form.

isuruf commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+  home: https://github.com/malthe/macfsevents
+  license: MIT
+  license_family: MIT
+  license_file: ''  # No license file provided in source repo
```
Can you ask upstream to include one with a license text so that we know which version of BSD is used?

pbronez replied:
Sure, opened an [issue](https://github.com/malthe/macfsevents/issues/38). Note that Pypi metadata says MIT while the README says BSD.

isuruf commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+about:
+  home: https://github.com/malthe/macfsevents
+  license: MIT
+  license_family: MIT
```
Repo says BSD

pbronez replied:
Yeah, Pypi metadata is inconsistent with the README. I've asked the upstream author [to clarify](https://github.com/malthe/macfsevents/issues/38)

isuruf commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+  license_family: MIT
+  license_file: ''  # No license file provided in source repo
+  summary: Thread-based interface to file system observation primitives.
+  description: ".. contents::\n\nOverview\n========\n\n.. role:: mod(emphasis)\n\n:mod:`MacFSEvents` is a Python library that provides thread-safe\ndirectory observation primitives using callbacks. It wraps\
```
This is too long and unnecessary. Can you make it smaller and readable?

pbronez replied:
Addressed in https://github.com/conda-forge/staged-recipes/pull/4219/commits/3a1e6fd731ec961d23e7127647bd335a4696d90e

isuruf commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,78 @@
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.python.org/packages/f4/fb/59f72719e339f6209997414c6d0b7e1e1f96900dede3ec1cc24008471cc2/MacFSEvents-0.7.tar.gz
```
Use the link as in the example recipe at https://github.com/conda-forge/staged-recipes/tree/master/recipes/example

pbronez replied:
Yes, I started from the example + conda skeleton. This failed to build due to 301 responses from Pypi. (Other packages were building fine -- just this one had a problem). I hardcoded a direct link and that seemed to solve the problem.

I'm not sure why I was hitting redirects for this package. Best I can figure it might be related to [this issue](https://github.com/LLNL/spack/issues/1939)

isuruf replied:
https://pypi.io/packages/source/M/MacFSEvents/MacFSEvents-0.7.tar.gz works for me.

pbronez commented:
Upstream clarified licensing as requested: https://github.com/malthe/macfsevents/commit/e99231e8382b6a6cbc769bfa17c44ea9c1a35c37 I updated to reflect this change in https://github.com/conda-forge/staged-recipes/pull/4219/commits/7988bef12d551e81776959fd9ffbfda3de29e74d

I reverted back to the example URL construction: https://github.com/conda-forge/staged-recipes/pull/4219/commits/7f9a00d24da417442edae90a742bb9de678305e1 PyPi isn't throwing 301s anymore, must have been a transient issue.

dougalsutherland commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,49 @@
+{% set name = "macfsevents" %}
```
If you change this to `MacFSEvents`, which is what the pypi name is, I think the download will start working.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/MacFSEvents```) and found it was in an excellent condition.

pbronez commented:
@dougalsutherland yup, that fixed it. Thanks.

dougalsutherland commented:
No need to rename the recipe directory; conda uses all-lowercase names by convention, no matter the PyPI casing. I *think* the feedstock creator will use the package `name`, but might as well change it back.....

In terms of the license, you should [package it manually](https://conda-forge.org/docs/meta.html#packaging-the-license-manually) and also file an issue upstream about it (or just send a PR, the needed changes are simple; for example, https://github.com/bokeh/colorcet/pull/6) then put a comment with the URL in the recipe.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/macfsevents```) and found it was in an excellent condition.

pbronez commented:
@dougalsutherland thanks for the review. I manually packaged LICENSE.txt and made an upstream [issue](https://github.com/malthe/macfsevents/issues/39) & [pull request](https://github.com/malthe/macfsevents/pull/40) to include LICENSE.txt in the standard build.

I also reverted the case change as requested.

dougalsutherland commented:
Hey, sorry, one more thing I missed before: because this compiles code, it should have the `toolchain` build requirement. I just added that; will merge assuming the CI still passes.

jakirkham commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,52 @@
+Mac OS X 10.5+ (Leopard)
+Python 2.7+
+
+This software was written by Malthe Borch <mborch@gmail.com>. The pyfsevents module by Nicolas Dumazet was used for reference."
```
nit: Would be nice if the `description` section were a bit more contained. Recommend using `description: |` and indenting text to fit underneath. This should improve readability. Here's a [good example]( https://github.com/conda-forge/willow-feedstock/blob/6add334d5a8d297b0b078b5f2433ce6a5de17f28/recipe/meta.yaml#L35-L38 ).

jakirkham commented on recipes/macfsevents/meta.yaml:
```diff
@@ -0,0 +1,52 @@
+build:
+  number: 0
+  skip: True  # [not osx]
+  script: python setup.py install  --single-version-externally-managed --record=record.txt
```
nit: Very minor suggestion. Seems there is an extra space between `install` and the first argument. Would suggest cutting it to one space.

scopatz commented:
The nits can be fixed up in the feedstock.
